gateway-ha/src/main/java/io/trino/gateway/ha/clustermonitor/BackendsMetricStats.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.clustermonitor;
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.airlift.log.Logger;
+import io.trino.gateway.ha.config.MonitorConfiguration;
+import io.trino.gateway.ha.config.ProxyBackendConfiguration;
+import io.trino.gateway.ha.router.GatewayBackendManager;
+import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
+import org.weakref.jmx.MBeanExporter;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Singleton
+public class BackendsMetricStats
+{
+    private static final Logger log = Logger.get(BackendsMetricStats.class);
+    public static final int DEFAULT_METRIC_REFRESH_SECONDS = 30;
+
+    private final MBeanExporter exporter;
+    private final GatewayBackendManager gatewayBackendManager;
+    private final MonitorConfiguration monitorConfiguration;
+    private Map<String, BackendClusterMetricStats> statsMap = new ConcurrentHashMap<>();
```

Comment:
I was going to suggest that we just use `exporter.getExportedObjects()` to check for which objects need to be (un)registered instead of maintaining a separate map, but since `MBeanExporter` uses [weak values](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ref/WeakReference.html), I don't think it will work -- they'll get garbage collected.

Assuming you agree, can we at least add a comment saying why we need to maintain this additional map?